preparata.yml

```diff
@@ -0,0 +1,48 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+# code id, physical, logical are all lower case
+code_id: preparata
+physical: bits
+logical: bits
+
+name: 'preparata'
+introduced: '\url{https://doi.org/10.1007/0-387-34799-2_39}'
+
+description: 'Preparata code is a \([[2^{m+1}-1, 2^{m+1}-2m-2, 5]]\) binary code where \(m\) is odd. The code is non linear and can exist for perimeters 
+for which there is no linear code. Extended preparata codes can be thought of as characteristic function of the tuple (X,Y) where X, Y belongs to \(\mathbb{F}_{2^m}\) along with the
+following properties:- (i) \(|X| = |Y| = 2^{m}\) (ii) \(\sum_{x \in X} x = \sum_{y \in Y} y \) (iii) \(\sum_{x \in X} x^{3} + (\sum_{x \in X} x)^{3}  = \sum_{y \in Y} y^{3} \). Preparata
+codes are then obtained by leaving out the coordinate in the zeroeth positions in the first half.'
+
+protection: 'Preparata codes are a class of non linear double error correcting codes.'
+
+features:
+
+  rate: 'The transmission rate is \(\frac{2^{m+1}-2m-2}{2^{m+1}-1}\).'
+
+  encoders:
+    - ''
+
+  decoders:
+    - 'Trellis diagrams employed for soft decision decoding. '
+
+notes:
+  - 'Preparate codes are uniformly packed and nearly perfect which implies that for any word \(textbf{u}\)' and (preparata) codebook C with \(d(\textbf{u}, C) > 2\), we have
+     that \(u\) has a distance 2 or 3 to exactly floor((2^{m+1}-1)/3) codewords.'  
+  - 'Preparata codes contain twice as many code words as the double-error-correcting BCH codes of the same length, which is the largest number of code words
+     possible for given length and distance. Thus, they are optimal\cite{https://doi.org/10.1016/S0019-9958(68)90874-7}.'
+  - 'Preparata codes are linear over Z4 with the Lee distance.'    
+
+relations:
+  parents:
+    - code_id: bch
```

Comment:
BCH is a relative not a parent

Reply:
Move to cousin. Parents are bits_into_bits and quaternary_over_z4